bugwarrior/services/azuredevops.py

```diff
@@ -0,0 +1,209 @@
+        return self.build_default_description(
+            title=self.record["fields"]["System.Title"],
+            url=self.get_processed_url(self.record["_links"]["html"]["href"]),
+            number=self.record["id"],
+            cls=self.record["fields"]["System.WorkItemType"].lower(),
+        )
+
+
+class AzureDevopsService(IssueService):
+    ISSUE_CLASS = AzureDevopsIssue
+    CONFIG_PREFIX = "ado"
+
+    def __init__(self, *args, **kw):
+        super(AzureDevopsService, self).__init__(*args, **kw)
+        self.host = self.config.get("host", "dev.azure.com")
+        self.PAT = self.config.get("PAT")
+        self.project = self.config.get("project")
+        self.org = self.config.get("organization")
+        self.client = AzureDevopsClient(
+            pat=self.PAT, project=quote(self.project), org=quote(self.org), host=self.host
+        )
+        self.query_filter = self.config.get("wiql_filter")
+
+    def get_query(self):
+        default_query = (
+            "SELECT [System.Id] FROM workitems WHERE [System.AssignedTo] = @me"
+        )
+        if self.query_filter:
+            default_query += f"AND {self.query_filter}"
+        list_of_items = self.client.get_work_items_from_query(default_query)
+        return list_of_items
+
+    def annotations(self, issue, issue_obj):
+        # Build Annotations based on comments and description
+        url = issue["_links"]["html"]["href"]
+        annotations = []
+        if self.annotation_comments:
+            comments = self.client.get_workitem_comments(issue)
+            if comments:
+                for comment in comments:
+                    try:
+                        name = comment["revisedBy"]["displayName"]
+                    except KeyError:
+                        name = comment["modifiedBy"]["displayName"]
+                    text = format_item(comment["text"])
+                    annotations.append((name, text))
+        return self.build_annotations(annotations, issue_obj.get_processed_url(url))
+
+    def issues(self):
+        issues = {}
+        issue_ids = self.get_query()
+        for issue_id in issue_ids:
+            issue = self.client.get_work_item(issue_id)
+            parent_title = self.client.get_parent_name(issue)
+            issue["ParentTitle"] = parent_title
+            issue_obj = self.get_issue_for_record(issue)
+            extra = {
+                "project": issue["ParentTitle"],
+                "annotations": self.annotations(issue, issue_obj),
+                "namespace": f"{self.org}\\{self.project}",
+            }
+            issue_obj.update_extra(extra)
+            yield issue_obj
+        pass
+
+    @classmethod
+    def validate_config(cls, service_config, target):
+        for option in ("PAT", "project", "organization"):
+            if option not in service_config:
+                die(f"[{target}] has no 'ado.{option}'")
+        IssueService.validate_config(service_config, target)
```

Comment:
Please use `super` here to access the base class (as the other services do).

Reply:
It would be my luck that the service I copied from is one of the only ones that doesn't do this 😆 

https://github.com/ralphbean/bugwarrior/blob/1650269b483c4abf197e3d65f4442eeb3802c8fd/bugwarrior/services/jira.py#L354